lib/record_store/provider/oracle_cloud_dns.rb

```diff
@@ -0,0 +1,195 @@
+        stdout.puts "Cannot build recrod for zone: #{zone}"
+      end
+
+      # Returns an array of the zones managed by provider as strings
+      def zones
+        client.list_zones(secrets['compartment_id']).data.map(&:name)
+      end
+
+      private
+
+      # Returns the version of the zone
+      def zone_version(zone)
+        client.get_zone(zone).data.version
+      end
+
+      # Creates a new record to the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # record - a kind of `Record`
+      def add(record, zone)
+        record_rdata = retrieve_rdata_based_on_record_type(record)
+        patch_add_record = [
+          OCI::Dns::Models::RecordOperation.new(
+            domain: record.fqdn,
+            rtype: record.type,
+            ttl: record.ttl,
+            rdata: record_rdata,
+            operation: 'ADD',
+          ),
+        ]
+
+        client.patch_zone_records(
+          zone,
+          OCI::Dns::Models::PatchZoneRecordsDetails.new(items: patch_add_record)
+        )
+      end
+
+      # Deletes an existing record from the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # record - a kind of `Record`
+      def remove(record, zone)
+        record_fqdn = Record.ensure_ends_without_dot(record.fqdn)
+        record_rdata = retrieve_rdata_based_on_record_type(record)
+        found_record = client.get_zone_records(
+          zone,
+          rtype: record.type,
+          zone_version: client.get_zone(zone).data.version,
+          domain: record_fqdn,
+        ).data.items.last
+
+        begin
+          record_hash = found_record.record_hash
+          patch_remove_record = [
+            OCI::Dns::Models::RecordOperation.new(
+              domain: record_fqdn,
+              record_hash: record_hash,
+              rtype: record.type,
+              ttl: record.ttl,
+              rdata: record_rdata,
+              operation: 'REMOVE',
+            ),
+          ]
+
+          client.patch_zone_records(
+            zone,
+            OCI::Dns::Models::PatchZoneRecordsDetails.new(items: patch_remove_record)
+          )
+        rescue NoMethodError
+          puts "An exception occurred while deleting changeset which doesn't exist"
+        end
+      end
+
+      # Updates an existing record in the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # id - provider specific ID of record to update
+      # record - a kind of `Record` which the record with `id` should be updated to
+      def update(id, record, zone)
+        record_fqdn = Record.ensure_ends_without_dot(record.fqdn)
+        record_rdata = retrieve_rdata_based_on_record_type(record)
+
+        # Retrieve all records that you want to keep before it's updated because it will overwrite
+        all_records = []
+        client.get_zone_records(zone, zone_version: zone_version(zone)).each do |response|
```

Comment:
Was that kept around as-is ?

Reply:
This should've been updated though, will be update with new commit!